src/System.Buffers/src/System/ByteSpan.cs

```diff
@@ -74,7 +75,8 @@ public int Length
 
         public ByteSpan Slice(int index)
         {
-            Precondition.Require(index < Length);
+            if (index >= Length)
+                return new ByteSpan(_data, 0);
```

Comment:
But you could accomplish that by allowing slicing to zero length, but not "past" that.